text/000-rustdoc-jump-to-definition.md

```diff
@@ -0,0 +1,178 @@
+
+So in short, this feature is split in two parts:
+
+ * Jump to the definition of an ident.
+ * Jump to an item documentation from the source view.
+
+> @GuillaumeGomez comment: I tried to explain why I think having jump to def links and "back to doc" links is important, but I feel like I'm explaining it in some weird and unclear ways... Any suggestion is very welcome.
+
+Adding this information directly into the documentation allows it to improve the documentation value: if you want to go a bit further what is written in the documentation (to see some implementation details for example), you can currently go to the source code. However, you're quickly limited to the current file if it's using private items not from this file.
+
+Once you found what you're looking for, having a link to an item's documentation from the source code allows to add the missing connection between the source code viewer and the doc pages.
+
+Another important note is that if documentation is split in multiple places or requires an external tool for the source code navigation, it's actually slowing down our users quite a lot.
+
+> @GuillaumeGomez comment: This is the "story" part. Is it ok to put it here or should it be somewhere else? Also: is it clear enough?
+
+A good example where this would be very helpful is for the [rust compiler source code](https://doc.rust-lang.org/1.57.0/nightly-rustc/src/rustc_middle/hir/map/mod.rs.html#872-877): even if a fully set up IDE or github, it's very hard to go around without previous knowledge or help from someone else. It's also very common to have undocumented items or to just want to see how something is implemented. With the "jump to definition", it already improved quite a lot the browsing experience by allowing to jump super quickly by simply clicking on a item. The missing part is now to go back from the source code into the item documentation to make it complete.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+When going to the source code pages (as you can see already in the [compiler documentation](https://doc.rust-lang.org/1.57.0/nightly-rustc/src/rustc_middle/hir/map/mod.rs.html#872-877)), you will be able to click on idents to go to their definition.
+
+Once at their definition, if you click on the item name, you will arrive on its documentation page (if it has one, otherwise nothing will happen).
+
+All items with documentation should have a link to their documentation page (like impl blocks, methods, trait/impl associated items, etc).
+
+An interesting thing to note: primitive types always link to documentation.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+It doesn't require javascript since it's only generating links, making it work with or without javascript enabled.
+
+So on the technical side, there is no big issues. You can see the first part of this RFC implemented [here](https://github.com/rust-lang/rust/pull/84176).
```

Comment:
issue: this doesn't really belong here, also language like "there are no big issues" is presupposing what the issues _are_

Reply:
Based on what is described in this RFC. I'll add this sentence to make it more clear.